New GUI for the WebBrowser

Falcort opened the pull request:
This PR change the design of the current WebBrowser into something more user friendly

Falcort commented:
@linev Rebase done

linev commented:
Good, once you want to merge it - please let me know

Falcort commented:
Maybe it is better for merge it after i found a solution for the SplitApp resize and move some stuff into the ToolHeader ?

bellenot commented:
> Maybe it is better for merge it after i found a solution for the SplitApp resize and move some stuff into the ToolHeader ?

Yes, let's try to fix it first (if it can be done quickly enough)

Falcort commented:
@linev This PR for the new GUI is rebase an can now be reviewed

linev commented:
I looked into the PR. There are two point.

1. You add "Browse..." button to load files using default file open dialog from the web browser. Does it make sense? We already have file browsing on left side. It can fully confuse users when they work with remote system. I propose to "hide" this functionality for now and provide in "Expert" mode. And one has to implement correspondent "Save as ..." button which allows to save text editor content into browser host system. As it done in JSROOT with saving png files:

https://github.com/root-project/jsroot/blob/master/scripts/JSRootPainter.v6.js#L3982-L3998

2. You are using ToolHeader from **tnt** library. This library is not included into ROOT openui5 package, therefore does not work properly in "offline" mode. Are there any other alternatives in libraries like **m** or **ui**? If not, I have to include **tnt** - it seems to be not that big.

linev commented:
Also some time ago we discuss functionality to resize left area in the SplitApp (where files list is shown). Did you found correspondent components?

Falcort commented:
1. This button was not added by me, it was meant to be use ro test the code editor before the fileDialog was created. But i can remove it if you want.

2. I will look into that but i don't think **m** or **ui** have such things, if i remember well, all the others solutions where depreciated

linev commented:
> This button was not added by me, it was meant to be use ro test the code editor before the fileDialog was created. But i can remove it if you want.

Just "hide" this button for the moment - it will be confusing to the people.

> I will look into that but i don't think m or ui have such things

Please check. If there are no alternatives - I will add tnt lib to ROOT openui5 package

Falcort commented:
> Also some time ago we discuss functionality to resize left area in the SplitApp (where files list is shown). Did you found correspondent components?

Yes and no, the component that you showed me on the demo was a very big and complex component that would have need a complete refactor of a lot of things

https://openui5.hana.ondemand.com/entity/sap.f.tutorial.fiori2/sample/sap.f.tutorial.fiori2.12 

The other solution that i found that would have been custom resize will not fit also because of the Tree table :

https://openui5.hana.ondemand.com/entity/sap.tnt.ToolPage/sample/sap.tnt.sample.ToolPage

So for now i don't think this is possible with a UI5 component, i thinks that we can recode the functionality of the SplitApp using just a Splitter, but it will be complexe for the Responsive

linev commented:
I saw much simpler example how MasterPage can be resized.

https://answers.sap.com/questions/12009929/is-it-possible-to-expand-the-width-of-master-page-.html

It is just an single parameter which set width of master page - one can add button in the ToolHeader to switch between default and wide mode.

Falcort commented:
Yes if you want to do that yes it is possible, but the best solution would be to me able to drag the master into the desired length
But yes i can implement those button if you want !

linev commented:
> But yes i can implement those button if you want !

It is simplest solution for now, later you can try do it better

bellenot commented:
> 1. You add "Browse..." button to load files using default file open dialog from the web browser

FYI, this button is there since day one...

linev commented:
> FYI, this button is there since day one...

I was ignoring it all the time :(
But one should disable it

Falcort commented:
@linev can I create a css file to move all the styles from the index to that new file ?

linev commented:
> @linev can I create a css file to move all the styles from the index to that new file ?

Yes, but be aware that location of such file will be: **"rootui5sys/browser/yourfile.css" ** if you put file in the same directory as index.html

Falcort commented:
@linev here is what i have done :
- Added a button that change the size of master to 40% of the screen.
- Added a button to revert back to original size
- I moved all the styles into a CSS file
- Removed the "Browse ..." button

I also did not found alternatives for **tnt**

linev commented:
There is side effect when resizing web browser. I saw situation when both buttons - "Enter full screen" and "Exit full screen" appears at the same time. Just toggle "full screen", then made web browser smaller (main page hides) and then again web browser bigger. Please check logic here. And also name "full screen" is misleading. Wider or thinner

linev commented:
I apply #4822, now new RBrowser with sap.tnt library should work

Falcort commented:
@linev i will debug that as soon as position, but this was part of an intended behavior.
As in portait mode the master take 80% of the screen making it bigger is not useful
So the goad was to hide those button and revert to normal the master when entering portrait move, but this is not behaving correctly
